### PR TITLE
For a exorbitant fee, you can purchase full machine beacons from cargo.

### PR DESCRIPTION
**Pull request by ArcaneMusic**
## About The Pull Request

Cargo can now order supply beacons crates. Supply beacons are similar to the roulette beacon spawner, in that I refactored those into a base type that just spawns a certain type of object with them now to make everyone's life easier going forward.

The supply beacon packs allows you to purchase high use machinery directly from cargo, at the cost of being exceptionally expensive compared to their base cost and effort to just build one.  Currently, these are limited to being machinery that spawns somewhere on the map, and with a basic amount of effort you can already obtain these with minimal effort on station, but as it stands, sometimes it's just a pain in the ass as a doctor to drag the CMO to the bridge, get them to research biotech, drag a scientist to the desk, get them to print you the appropriate parts, then find the raw supplies and materials back to medbay to upgrade your department with 1-2 machines.

Now, can use your "overflowing" funds to buy replacements or spares of some of the most common machines on the station, within perfectly legal channels.

## Why It's Good For The Game

Average player finds at the end of a highpop shift trend to be about 1100 to 3000 credits if heads are involved in a round. This leaves most of the machines on this list as something a single player would need to bust ass to purchase within a 90 minute round, but serves as a nice suplementary goal for player funds. People want more ways to spend money, and in spite of the costs being exceptionally high, these make for acceptably difficult goals for players to strive towards using private funds to purchase.

Pairs well with departmental ordering in #53881, as each of these will need a departmental can_view variable assigned for each supply crate in order to prevent omni-department singularies from occurring more than they already do.

Original idea was from a discussion with maintainers, but the numbers are 100% subject to change and I'm more than willing to raise prices in areas if seen fit.

## Changelog
:cl:
add: The cargo department now offers supply beacons to be purchased, which will deploy fully built machines to the crew at a high cost.
/:cl:

### CODE REVIEW

**Comment by optimumtact** [comment]
what a stupid idea

**Comment by Tlaltecuhtli** [comment]
will the civilian bounty gains be increased? (after the nerf people stopped using them) as now cargo will need  10 times more bounties to reach what you would get before

File: code/game/objects/items/beacons.dm [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+
+///Food processor
+/obj/item/sup_beacon/processor
+	name = "food processor beacon"
+	desc = "N.T. approved food processor beacon, toss it down and you'll have a fully ready food prep machine. It slices, it dices, it even makes julianne fries!"
+	beacon_contents = /obj/machinery/processor
+
+/obj/item/sup_beacon/hydroponics
+	name = "hydroponics tray beacon"
+	desc = "N.T. approved hydroponics tray beacon, toss it down and you'll have a fully ready hydroponics tray. Grow's just about anything the stomache desires."
```
Comment (MrDoomBringer) [inline code/game/objects/items/beacons.dm:64]:
```suggestion
	desc = "N.T. approved hydroponics tray beacon, toss it down and you'll have a fully ready hydroponics tray. Grows just about anything the stomache desires."
```

File: code/game/objects/items/beacons.dm [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+	beacon_contents = /obj/machinery/hydroponics
+
+//******Medical******
+
+///Statis Bed
+/obj/item/sup_beacon/stasis
+	name = "stasis bed beacon"
+	desc = "N.T. approved stasis bed beacon, toss it down and you'll have a fully ready statis bed. Turn anywhere into an operating theator! Or don't!"
```
Comment (MrDoomBringer) [inline code/game/objects/items/beacons.dm:72]:
```suggestion
	desc = "N.T. approved stasis bed beacon, toss it down and you'll have a fully ready stasis bed. Turn anywhere into an operating theator! Or don't!"
```

File: code/game/objects/items/beacons.dm [hunk 3]
```diff
@@ -0,0 +1,113 @@
...
+	beacon_contents = /obj/machinery/stasis
+
+/obj/item/sup_beacon/chem_dispenser
+	name = "chemistry dispenser beacon"
+	desc = "N.T. approved chemistry dispenser beacon, toss it down and you'll have a fully ready chemistry dispenser. Faster than starting an inter-departmental warzone!"
+	beacon_contents = /obj/machinery/chem_dispenser
+
+/obj/item/sup_beacon/chem_heater
+	name = "chemistry heater beacon"
+	desc = "N.T. approved chemistery heater beacon, toss it down and you'll have a fully ready chemistry heater. Infinitely safer than doing it on the stove."
```
Comment (MrDoomBringer) [inline code/game/objects/items/beacons.dm:82]:
```suggestion
	desc = "N.T. approved chemistry heater beacon, toss it down and you'll have a fully ready chemistry heater. Infinitely safer than doing it on the stove."
```

File: code/game/objects/items/beacons.dm [hunk 4]
```diff
@@ -0,0 +1,113 @@
...
+	beacon_contents = /obj/machinery/chem_heater
+
+/obj/item/sup_beacon/chem_master
+	name = "chemistry master beacon"
```
Comment (MrDoomBringer) [inline code/game/objects/items/beacons.dm:86]:
```suggestion
	name = "chem-master beacon"
```

File: code/game/objects/items/beacons.dm [hunk 5]
```diff
@@ -0,0 +1,113 @@
...
+	desc = "N.T. approved chemistry master beacon, toss it down and you'll have a fully ready chemistry master. Even we don't know where all those little bottles come from."
```
Comment (MrDoomBringer) [inline code/game/objects/items/beacons.dm:87]:
```suggestion
	desc = "N.T. approved chem-master beacon, toss it down and you'll have a fully ready ChemMaster 3000. Even we don't know where all those little bottles come from."
```

File: code/game/objects/items/beacons.dm [hunk 6]
```diff
@@ -0,0 +1,113 @@
...
+	beacon_contents = /obj/machinery/chem_master
+
+//******Engineering******
+
+/obj/item/sup_beacon/emitter
+	name = "emitter beacon"
+	desc = "N.T. approved emitter beacon, toss it down and you'll have a fully ready laser emitter. Turn those rookie engine numbers into some BIG rookie numbers."
```
Comment (MrDoomBringer) [inline code/game/objects/items/beacons.dm:94]:
```suggestion
	desc = "N.T. approved emitter beacon, toss it down and you'll have a fully ready laser emitter. Turn those rookie engine numbers into some BIG engine numbers."
```

File: code/game/objects/items/beacons.dm [hunk 7]
```diff
@@ -0,0 +1,113 @@
...
+	beacon_contents = /obj/machinery/power/emitter
+
+/obj/item/sup_beacon/freezer
+	name = "atmospherics freezer beacon"
+	desc = "N.T. approved freezer beacon, toss it down and you'll have a fully ready atmospherics freezer. Keep cool, "
```
Comment (MrDoomBringer) [inline code/game/objects/items/beacons.dm:99]:
```suggestion
	desc = "N.T. approved freezer beacon, toss it down and you'll have a fully ready atmospherics freezer. Stay frosty. Note that with some simple user service, this machine can be turned into a heater."
```

File: code/game/objects/items/beacons.dm [hunk 8]
```diff
@@ -0,0 +1,113 @@
...
+//******Security******
+
+/obj/item/sup_beacon/weapon_recharger
+	name = "weapon recharger beacon"
+	desc = "N.T. approved weapon recharger beacon, toss it down and you'll have a fully ready recharger. Makes short work of shorter stunpod cells."
+	beacon_contents = /obj/machinery/recharger
+
+/obj/item/sup_beacon/cameras
+	name = "camera console beacon"
+	desc = "N.T. approved camera console beacon, toss it down and you'll have a fully ready camera console. Home survailance... anywhere!"
```
Comment (MrDoomBringer) [inline code/game/objects/items/beacons.dm:111]:
```suggestion
	desc = "N.T. approved camera console beacon, toss it down and you'll have a fully ready camera console. Home surveillance... anywhere!"
```

File: code/modules/cargo/packs.dm [hunk 9]
```diff
@@ -2418,6 +2418,102 @@
 		cardpacktype = pick(subtypesof(/obj/item/cardpack))
 		new cardpacktype(C)
 
+//////////////////////////////////////////////////////////////////////////////
+////////////////////////// Pre-Built Machines ////////////////////////////////
```
Comment (MrDoomBringer) [inline code/modules/cargo/packs.dm:2422]: IMO  you should really switch the descriptions for the supply pack names and the item names. Usually, the supply pack names - as listed in the cargo console - are a bit more whimsical and advertisement-like, which seems to be the reverse of what's going on here.

**Review by MrDoomBringer — CHANGES_REQUESTED** [review]
based pr but that's a lot of typos lmao

File: code/game/objects/items/beacons.dm [hunk 10]
```diff
@@ -0,0 +1,113 @@
+///The all new home for beacons that spawn objects, structures, atoms, whatever to the station via supply pods.
+/obj/item/sup_beacon
+	name = "beacon"
+	desc = "N.T. approved supply beacon, but this one shouldn't exist!"
```
[the remark this answers is not in the document]
Comment (Tlaltecuhtli) [reply code/game/objects/items/beacons.dm:4]: make a github report*

[the remark this answers is not in the document]
Comment (MrDoomBringer) [reply code/game/objects/items/beacons.dm:4]: edited

File: code/game/objects/items/beacons.dm (hunk 2 above)
Comment (TheBonded) [reply code/game/objects/items/beacons.dm:72]: Isn't it theater or theatre? No o I believe

Comment (MrDoomBringer) [reply code/game/objects/items/beacons.dm:72]: yup lol - edited

**Comment by ExcessiveUseOfCobblestone** [comment]
> sometimes it's just a pain in the ass as a doctor to drag the CMO to the bridge, get them to research biotech, drag a scientist to the desk, get them to print you the appropriate parts, then find the raw supplies and materials back to medbay to upgrade your department with 1-2 machines.

Im really confused since this seems like a perfect time to use money. Why is the answer to this "I need to generate a new outlet code-wise in cargo" and not "make a bounty board and let people know they can get cash if they help me out"?

**Comment by Krysonism** [comment]
> 
> 
> > sometimes it's just a pain in the ass as a doctor to drag the CMO to the bridge, get them to research biotech, drag a scientist to the desk, get them to print you the appropriate parts, then find the raw supplies and materials back to medbay to upgrade your department with 1-2 machines.
> 
> Im really confused since this seems like a perfect time to use money. Why is the answer to this "I need to generate a new outlet code-wise in cargo" and not "make a bounty board and let people know they can get cash if they help me out"?

Or buy stock parts from the stock part vendor.

**Comment by ArcaneMusic** [comment]
Based on maintainer input by cobby, I'll close this for now, but may reopen with just the code improvement for beacons so that they stop being so shit.

**Pull request closed without merging**